drivers/platform/maxim/gpio_extra.h

```diff
@@ -57,25 +57,20 @@ extern const struct gpio_platform_ops max_gpio_ops;
  */
 extern const struct irq_platform_ops max_gpio_irq_ops;
 
-/**
- * @struct gpio_irq_config
- * @brief maxim configuration descriptor for irq operations
- */
-struct gpio_irq_config {
-	enum irq_trig_level mode;
+enum max_pad_config {
+	NO_OS_PAD_NONE,
```

Comment:
I think the naming is a bit weird.
NO_OS_* would be the prefix for something in no-os/gpio.h but here I guess the prefix should be MAX_*

Reply:
On the other hand, I agree we should have a NO_OS_GPIO_PULLUP / PULLDOWN etc. enum somewhere in gpio.h.